pinot-controller/src/main/java/com/linkedin/pinot/controller/ControllerConf.java

```diff
@@ -65,6 +68,7 @@
   private static final int DEFAULT_JERSEY_ADMIN_PORT = 21000;
   private static final String DEFAULT_ACCESS_CONTROL_FACTORY_CLASS =
       "com.linkedin.pinot.controller.api.access.AllowAllAccessFactory";
+  private static final long DEFAULT_SEGMENT_COMMIT_TIMEOUT_MILLIS = 10 * 60 * 1000; // 10 minutes
```

Comment:
This is a very long time to store a segment, but I guess we are being conservative here. That is fine. In that case, perhaps the sender (hadoop job) should set the socket read timeout to this number, maybe. Not sure if we can find a way to send 100 continue response as we get the segment... something to think about.